bigml/sourcehandler.py

```diff
@@ -86,6 +88,21 @@ def _create_remote_source(self, url, args=None):
         body = json.dumps(create_args)
         return self._create(self.source_url, body)
 
+    def _create_inline_source(self, src_obj, args=None):
+        """Create source from inline data
+
+        """
+        create_args = {}
+        if args is not None:
+            create_args.update(args)
+        if 'source_parser' in create_args:
```

Comment:
Is there a special reason for this 'source_parser' handling? args is supposed to contain a dictionary with all the nested structure you need to specify all changes. The final json.dumps in line 183 should take care of the JSON string encoding at the end, isn't it?